src/components/log/LiveLog.tsx

```diff
@@ -0,0 +1,128 @@
+/* Pi-hole: A black hole for Internet advertisements
+ * (c) 2019 Pi-hole, LLC (https://pi-hole.net)
+ * Network-wide ad blocking via your own hardware.
+ *
+ * Web Interface
+ * Live Log component
+ *
+ * This file is copyright under the latest version of the EUPL.
+ * Please see LICENSE file for your rights under this license. */
+
+import React, { Component } from "react";
+import { animateScroll } from "react-scroll";
+import { WithAPIData } from "../common/WithAPIData";
+import { getTimeFromTimestamp } from "../../util/dateUtils";
+import api from "../../util/api";
+import { Input, Container, Row, Col } from "reactstrap";
+import { WithTranslation, withTranslation } from "react-i18next";
+
+export interface LiveLogProps {
+  log: Array<{
+    timestamp: number;
+    message: string;
+  }>;
+}
+
+interface LiveLogState {
+  log: Array<{
+    timestamp: number;
+    message: string;
+  }>;
+}
+
+let nextId = 0;
+class LiveLog extends Component<LiveLogProps & WithTranslation, LiveLogState> {
+  static getDerivedStateFromProps(
+    state: LiveLogState,
+    props: LiveLogProps
+  ): LiveLogState {
+    return {
+      log: [...props.log, ...state.log]
+    };
+  }
+
+  state: LiveLogState = { log: [] };
+  scrollEnabled: boolean = true;
+
+  componentDidUpdate() {
+    if (this.scrollEnabled) {
+      this.scrollToBottom();
+    }
+  }
+
+  scrollToBottom() {
+    animateScroll.scrollToBottom({
+      containerId: "output",
+      duration: 250,
+      delay: 0,
+      isDynamic: true
+    });
+  }
+
+  checkBoxChange() {
+    this.scrollEnabled = !this.scrollEnabled;
+    this.scrollToBottom();
+  }
+
+  render() {
+    const outputStyle = {
+      width: "100%",
+      height: "648px"
+    };
+
+    const { t } = this.props;
+
```

Comment:
I noticed that the log is empty when you click on the nav link:
![Screenshot at 2019-12-03 13-46-32](https://user-images.githubusercontent.com/16748619/70052460-9dec1900-15d3-11ea-9357-b6ca235fd14a.png)

![Screenshot at 2019-12-03 13-46-47](https://user-images.githubusercontent.com/16748619/70052360-6aa98a00-15d3-11ea-9c19-a06b1ae55734.png)

The web interface requests data with the latest `nextId` in this case, hence, FTL does not send anything. However, you also seem to have lost the history of log lines here.

I'd recommend to always request without `nextId` when showing the page and only use `nextId` during the polling.

This can be achieved with the following:
```suggestion

    if (this.props.log.length === 0) {
      nextId = 0;
    }

```